app/controllers/events_controller.rb

```diff
@@ -0,0 +1,38 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :fetch_event_details, only: [:new, :create]
+
+  def index
+    @events = current_user.events
+    @owned_events = current_user.owned_events
+  end
+
+  def show
+    @event = current_user.events.find_by(id: params[:id])
+    @event = current_user.owned_events.find_by(id: params[:id]) if @event.blank?
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @event = current_user.owned_events.build(event_params.merge(image_url: @image))
+    pry binding
+    if @event.save
+      redirect_to event_path(@event)
+    else
+      render :new
+    end
+  end
+
+  private
+  def event_params
+    params.require(:event).permit(:name, :date)
+  end
+
+  def fetch_event_details
```

Comment:
No need to use it as `before_action`.

You can just define it as private method:
```ruby
 @event = current_user.owned_events.build(event_params.merge(image_url: fetch_image_url))

private

def fetch_image_url
  NasaService.new.get_image_url
end 
```